src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -65,6 +65,61 @@ class LogicValueActionSerializer(serializers.Serializer):
     )
 
 
+class SynchronizeChildrenVariableMappingSerializer(serializers.Serializer):
+    property = serializers.CharField(
+        required=True, label=_("Property that needs mapping.")
+    )
+    component_key = FormioVariableKeyField(
+        required=True,
+        label=_("Key of the editgrid's child component."),
+    )
+
+
+class SynchronizeChildrenActionConfigSerializer(serializers.Serializer):
+    source_variable = FormioVariableKeyField(
+        required=True,
+        allow_blank=False,
+        label=_("Key of the form variable that will be used as the source."),
+    )
+    destination_variable = FormioVariableKeyField(
+        required=True,
+        allow_blank=False,
+        label=_("Key of the form variable that will be used as the destination."),
+    )
+    children_mappings = SynchronizeChildrenVariableMappingSerializer(
+        many=True,
+        label=_("Synchronize children mapping"),
+    )
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+        mappings = attrs.get("children_mappings")
+
+        if not mappings:
+            raise serializers.ValidationError(
+                {"children_mappings": _("At least one mapping is required.")}
+            )
```

Comment:
`rest_framework.serializers.ListSerializer` (which gets used if you do `Serializer(many=True)` has a keyword argument `min_length` for this, so you don't need to write this validation yourself.

Reply:
As discussed with Viktor as well, I chose the manual validation because this provides a `nonFieldError` which imo is not the case. But if you disagree with that I can update it and refactor the code to handle the nonField errors as well.

https://github.com/open-formulieren/open-forms/pull/5496#discussion_r2333266196